src/main/java/core/basesyntax/utils/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.utils;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+
+    public Figure getRandomFigure(ColorSupplier colorSupplier) {
```

Comment:
ColorSupplier must be class field
```suggestion
    public Figure getRandomFigure() {
```